pkg/apis/apps/validation/validation.go

```diff
@@ -128,6 +130,12 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path, op
 
 	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(spec.Replicas), fldPath.Child("replicas"))...)
 	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(spec.MinReadySeconds), fldPath.Child("minReadySeconds"))...)
+	if utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetStartOrdinal) {
```

Comment:
we should not be checking feature gates here... if the field is non-nil, always validate it

please open a PR to fix this and backport it to 1.26 ahead of beta promotion

Reply:
same comment applies to other things in the validation.go file checking feature enablement

Reply:
Thanks for catching. Opened https://github.com/kubernetes/kubernetes/issues/116025 to track fix and 1.26 backport.